crates/icrate/examples/metal.rs

```diff
@@ -0,0 +1,344 @@
+#![deny(unsafe_op_in_unsafe_fn)]
+
+use core::ptr::NonNull;
+
+use icrate::{
+    AppKit::{
+        NSApplication, NSApplicationActivationPolicyRegular, NSApplicationDelegate,
+        NSBackingStoreBuffered, NSWindow, NSWindowStyleMaskClosable, NSWindowStyleMaskResizable,
+        NSWindowStyleMaskTitled,
+    },
+    Foundation::{
+        ns_string, NSDate, NSNotification, NSObject, NSObjectProtocol, NSPoint, NSRect, NSSize,
+        NSString,
+    },
+    Metal::{
+        MTLCommandBuffer, MTLCommandEncoder, MTLCommandQueue, MTLCreateSystemDefaultDevice,
+        MTLDevice, MTLDrawable, MTLLibrary, MTLPrimitiveTypeTriangle, MTLRenderCommandEncoder,
+        MTLRenderPipelineDescriptor, MTLRenderPipelineState,
+    },
+    MetalKit::{MTKView, MTKViewDelegate},
+};
+use objc2::{
+    declare::{Ivar, IvarDrop},
+    declare_class, msg_send, msg_send_id,
+    mutability::InteriorMutable,
+    rc::Id,
+    runtime::ProtocolObject,
+    ClassType,
+};
+
+#[repr(C)]
+pub struct SceneProperties {
+    pub time: f32,
+}
+
+#[repr(C)]
+pub struct VertexInput {
+    pub position: Position,
+    pub color: Color,
+}
+
+#[repr(C)]
+pub struct Position {
+    pub x: f32,
+    pub y: f32,
+    pub z: f32,
+}
+
+#[repr(C)]
+pub struct Color {
```

Comment:
Please note that this has the same ABI as `MTLPackedFloat3` (also goes for `Position`).

Reply:
Oh, I see, I missed those in the definitions. But do you want me to just make a comment about that or actually use them somehow? Since they don't seem to have any public fields or conversion traits, I don't see a safe way to use those directly in this case (i.e., wouldn't we need to do some transmuting or accessing elements at memory offsets unsafely?)

Reply:
> Since they don't seem to have any public fields or conversion traits

I think that's something to fix, but for now just commenting that they're the same is enough.